docs/experiments.rst

```diff
@@ -0,0 +1,318 @@
+the experiment as follows:
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to specify used builds for experiments in the experiments.yml file.
+
+   experiments:
+     - name: experiment1
+       args: ['<name_of_executable_of_build1>', ...]
+       used_builds: [build1]
+       ...
+
+In this way simexpal will check the :ref:`installation directory <BuildDirectories>` and the ``extra_paths``
+of the builds specified in ``used_builds`` for the executable. If a build
+:ref:`requires other builds <DependentBuilds>` and they are properly specified in the ``requires`` key, then
+simexpal will also check the installation directories and ``extra_paths`` of those builds.
+
+.. _ExperimentsRepeat:
+
+Repeat
+------
+
+Sometimes it might be useful to validate experiment results by repeating the experiment. In order to
+avoid duplicating an ``experiments`` entry we can use the
+
+- ``repeat``: integer - number of times an experiment is repeated
+
+key. To repeat an experiment twice we define our ``experiments`` stanza as follows:
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to specify repetitions for experiments in the experiments.yml file.
+
+   experiments:
+     - name: experiment1
+       ...
+       repeat: 2
+
+The default value of ``repeat`` is ``1``.
+
+Timeout
+-------
+
+It is possible to set a timer for experiments. Once the timer expires simexpal will terminate the
+experiment. To set the time we use the
+
+- ``timeout``: integer - timeout in seconds
+
+key. For example we can set the timer for an experiment to two hours as follows:
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to specify a timer for experiments in the experiments.yml file.
+
+   experiments:
+     - name: experiment1
+       ...
+       timeout: 7200
+
+.. _ExperimentsSettingEnvironmentVariables:
+
+Setting Environment Variables
+-----------------------------
+
+When using APIs like `OpenMP <https://www.openmp.org/spec-html/5.0/openmp.html>`_ it is sometimes
+necessary to specify settings as environment variables. Thus, simexpal supports setting environment
+variables in experiments by specifying the
+
+- ``environ``: dictionary of (environment variable, value)-pairs
+
+key. For example you can specify the ``OMP_NUM_THREADS`` environment variable as follows:
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to specify environment variables for experiments in the experiments.yml file.
+
+   experiments:
+     - name: experiment1
+       args: ...
+       ...
+       environ:
+         OMP_NUM_THREADS: 2
+      - name: experiment2
+        args: ...
+        ...
+        environ:
+         OMP_NUM_THREADS: 4
+
+Slurm
+-----
+
+.. _ExperimentsSupportedSlurmArgs:
+
+sbatch: ``--ntasks-per-node``, ``-c``, ``-N``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When using a job scheduler like `Slurm <https://slurm.schedmd.com/overview.html>`_ it might be
+useful to benchmark your algorithms using different node/cpu settings.
```

Comment:
simexpal is not only for benchmarking algorithms but for running empirical software experiments in general.

Reply:
changed "benchmark your algorithms" to "run your software"